crates/mun_codegen/src/ir/types.rs

```diff
@@ -7,6 +8,10 @@ impl<'ink> TransparentValue<'ink> for abi::Guid {
     fn as_target_value(&self, context: &IrValueContext<'ink, '_, '_>) -> Value<'ink, Self::Target> {
         self.0.as_value(context)
     }
+
+    fn as_bytes_and_ptrs(&self, _: &IrValueContext<'ink, '_, '_>) -> Vec<BytesOrPtr<'ink>> {
```

Comment:
I think the function name is a bit weird. It returns a bytes *or* ptr, but the function is called bytes *and* ptr.

Reply:
It returns a vector of `BytesOrPtr`, which means it can contain both. We want pointers to remain pointers; anything else needs to be converted to bytes. That makes it a function that converts to bytes and pointers; hence the name.

Reply:
👍